src/main/java/com/the/dark/side/crew/fejsbuk/service/impl/LikeServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+package com.the.dark.side.crew.fejsbuk.service.impl;
+
+import com.the.dark.side.crew.fejsbuk.mapper.LikeMapper;
+import com.the.dark.side.crew.fejsbuk.model.LikeEntity;
+import com.the.dark.side.crew.fejsbuk.model.dto.LikeDto;
+import com.the.dark.side.crew.fejsbuk.repository.LikeRepository;
+import com.the.dark.side.crew.fejsbuk.service.LikeService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+
+@Service
+@RequiredArgsConstructor
+public class LikeServiceImpl implements LikeService {
+
+    private final LikeRepository likeRepository;
+    private final LikeMapper likeMapper;
+
+    @Override
+    public LikeDto addLike(LikeDto likeDto) {
+        LikeEntity likeEntity = likeMapper.toEntity(likeDto);
+        if(likeRepository.existsByUserEntityAndPostEntity(likeEntity.getUserEntity(), likeEntity.getPostEntity())) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Already liked.");
+        } else likeRepository.save(likeEntity);
+        return likeMapper.toDto(likeEntity);
+    }
+
+    @Override
+    public void removeLike(long likeId) {
+        if(likeRepository.existsById(likeId)) {
+            likeRepository.deleteById(likeId);
+        } else throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Like " + likeId + " does not exist.");
```

Comment:
use { }

Reply:
https://stackoverflow.com/questions/12193170/whats-the-purpose-of-using-braces-i-e-for-a-single-line-if-or-loop